### ISSUE TITLE
Slow Training Progress

### ISSUE THREAD

**Issue by Bai-YT** [hirokiadachi/Customized-Adversarial-Training#1]
Dear hirokiadachi-san,

Thank you for your work and for sharing this implementation. I am now trying to reproduce the results, but 41 epochs into the training, I am still only getting a ~20% robust accuracy (natural accuracy is ~87%). I am hoping to ask if this is expected. Do you remember seeing the robust accuracy increase after more epochs (i.e., after reducing the learning rate at epoch 80), or is there likely a bug in my code (I made some minor modifications for compatibility with the newest PyTorch)? Really appreciate your help!

Sincerely,
Yatong

**Comment by Bai-YT** [comment]
This seems to be normal for this method. The robust accuracy increases after decreasing the learning rate.

**Comment by hirokiadachi** [comment]
Hi, I'm sorry for the delayed response. That's correct! The robust accuracy significantly increased after implementing the learning rate decay. I'll attach my training accuracy curve.
![W B Chart 12_4_2023, 9_48_31 PM](https://github.com/hirokiadachi/Customized-Adversarial-Training/assets/27120804/419b86c2-0c6d-4353-a154-b15eac174b17)
![W B Chart 12_4_2023, 9_48_44 PM](https://github.com/hirokiadachi/Customized-Adversarial-Training/assets/27120804/18f870c7-8fd0-4c75-a49b-d26c4f0fb9d5)

Additionally, please confirm I fixed the bugs in the inner maximization process. I needed to set the inner maximization model to evaluation mode.

**Comment by Bai-YT** [comment]
Thanks for the response! Are the attached figures obtained after fixing the evaluation mode problem? Would you mind sharing the model checkpoint after the bug fix? I sincerely appreciate your assistance!

**Comment by hirokiadachi** [comment]
No, these figures are the results before the bug fix. You can download the model checkpoint after the bug fix (https://drive.google.com/file/d/16YzuLhloAp1UZv99NDsm-TSEKiMwQINs/view?usp=sharing). This model is trained on the CIFAR-10 dataset and includes the last/best model.

**Comment by Bai-YT** [comment]
Thank you very much for sharing the model checkpoint. You are the best!
